test: kheap: add testcase to improve coverage

yingmingx opened the pull request:
Add a testcase to prove the thread can be waited on waitq when
there isn't enough space on heap.

Signed-off-by: Ying ming <mingx.ying@intel.com>

LeiW000 commented on tests/kernel/mem_heap/k_heap_api/src/test_kheap_api.c:
```diff
@@ -105,9 +119,38 @@ void test_k_heap_free(void)
+/**
+ * @brief Validate the k_heap support wait between different threads.
+ *
+ * @details In main thread alloc a buffer from the heap, than run the
```
Should "that" be "then"?

yingmingx replied:
Yes, it should be then here. Thanks.

LeiW000 commented:
@enjiamai, please take look at the PR.

enjiamai commented on tests/kernel/mem_heap/k_heap_api/src/test_kheap_api.c:
```diff
@@ -105,9 +119,38 @@ void test_k_heap_free(void)
  * @brief Validate allocation and free heap memory in isr context.
  *
+ * @details The test validates k_heap_alloc() in isr context, the timeout
+ * para should be K_NO_WAIT, because this situation isn't allow to wait.
```
is para a typo or intentional?

yingmingx replied:
Thank you for you careful review. I'll changed this word to param.

enjiamai reviewed (approved):
Looks fine. No other questions.

Commit:
test: kheap: add testcase to improve coverage

Add a testcase to prove the thread can be waited on waitq when
there isn't enough space on heap.

Signed-off-by: Ying ming <mingx.ying@intel.com>
